src/convert.rs

```diff
@@ -284,7 +266,7 @@ pub fn elf_to_tbf(
     let mut writeable_flash_regions_count = 0;
 
     for s in &sections_sort {
-        let section = &input.sections[s.0];
+        let section = &elf_file.sections[s.0];
```

Comment:
(sorry, github won't let me put the comment where I want)

We should move populating the `rel_sections` array to after the segment loop before. Basically, we no longer need to "guess" which `.rel.X` sections to add, we should know based on which sections were in loaded segments.

Reply:
What's your suggestion? During the program header loop to detect which sections belongs to the segments?